kong/runloop/balancer.lua

```diff
@@ -367,10 +367,31 @@ do
         checks.active.unhealthy.interval = 0
       end
 
+      local ssl_cert, ssl_key
+      if upstream.client_certificate then
+        local get_certificate = require("kong.runloop.certificate").get_certificate
+        local cert, err = get_certificate(upstream.client_certificate)
+        if not cert then
+          log(ERR, "unable to fetch upstream client TLS certificate ",
+              upstream.client_certificate.id, ": ", err)
+          return nil, err
+        end
+        ssl_cert = cert.cert
+        ssl_key = cert.key
```

Comment:
`cert.cert` and `cert.key` are `cdata` objects but the healthchecker expects strings. Will this work? I think the healthchecker should support `cdata` objects passed in directly as well.

Reply:
Nice catch! This was fixed in the healthchecker library, thanks!